SLRemoting/SLRESTContract.m

```diff
@@ -173,18 +173,23 @@ - (NSString *)patternForMethod:(NSString *)method {
 }
 
 - (NSString *)urlWithPattern:(NSString *)pattern
-                  parameters:(NSDictionary *)parameters {
+                  parameters:(NSMutableDictionary *)parameters {
     NSParameterAssert(pattern);
 
     if (!parameters) {
         return pattern;
     }
 
-    NSString __block *url = pattern;
+    NSString *url = pattern;
 
-    [parameters enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
-        url = [url stringByReplacingOccurrencesOfString:[NSString stringWithFormat:@":%@", key] withString:[NSString stringWithFormat:@"%@", obj]];
-    }];
+    for (NSString *key in parameters.allKeys) { // create a copy of allKeys to mutate parameters
+        NSString *keyPattern = [NSString stringWithFormat:@":%@", key];
+        if ([url rangeOfString:keyPattern].location != NSNotFound) {
```

Comment:
I would personally use a reverse logic here and reduce the level of nesting of the code below.

```objc
if ([url rangeOfString:keyPattern].location == NSNotFound) continue;
// etc.
```